dvc/repo/__init__.py

```diff
@@ -426,13 +435,14 @@ def find_outs_by_path(self, path, outs=None, recursive=False):
             outs = [out for stage in self.stages for out in stage.outs]
 
         abs_path = os.path.abspath(path)
+        path_info = PathInfo(abs_path)
         is_dir = self.tree.isdir(abs_path)
 
         def func(out):
-            if out.scheme == "local" and out.fspath == abs_path:
+            if out.scheme == "local" and path_info.isin_or_eq(out.path_info):
```

Comment:
Good question! Right now it will have a side effect of `dvc get ... dir/subdir` bringing you the whole dir. I was thinking about supporting get/import right away too, but better to handle it separately. So I'll add a `strict` flag or something and will only use it in `collect_granular` for now. Thank you for the heads up!